M2Mqtt/MqttClient.cs

```diff
@@ -2455,6 +2487,47 @@ internal bool Find(object item)
 
             }
         }
+
+        internal MqttMsgPublish ComposeMqttMsgPublish(
+            string topic,
+            byte[] message,
+            string contentType,
+            ArrayList userProperties,
+            MqttQoSLevel qosLevel,
+            bool retain)
+        {
+
+            MqttMsgPublish mqttMsgPublish = new(
+                topic,
+                message,
+                false,
+                qosLevel,
+                retain)
+            {
+                MessageId = GetMessageId()
+            };
+
+            mqttMsgPublish.ContentType = contentType;
```

Comment:
and you can skip the next line and the new added user properties is MQTT is not v5.0, little optimization but why not :-)

Reply:
Same goes for accepting ContentType